CeeFM/Decideify
# ContentCarousel.js refactor to allow for cycling through suggestions from first page to last page and vice versa!

**Description**
Carousels added to cycle through saved suggestions rather than showing them all at once on the mylibrary.js component and page! Pressing the previous button on the first page of results and then takes you to the last page of results in the carousel - where as before it was working when the total list of saved suggestions was not cleanly divisible by 5, but if it was cleanly divisible by 5, it took you to a blank set of results before hitting the last page - hooray for frustrating logic

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- Log in as coolin@gmail.com with password thistbh
- Every content type should show as a carousel in the My Library page! (If you don't have any saved suggestions, a message encouraging you to add suggestions to see them rendered should appear instead)
- On the first set of results shown in any carousel you should be able to press the back / previous button on the carousel and be shown the last set of results.
- Similarly, on the last set of results, you should be able to press the forward / next button on the carousel to be shown the first set of results again.

Checklist
